woodwork/tests/accessor/test_table_accessor.py

```diff
@@ -844,3 +844,96 @@ def test_is_valid_schema(sample_df):
 
     different_dtype_df = schema_df.astype({'id': 'Int64'}, copy=True)
     assert not _is_valid_schema(different_dtype_df, schema)
+
+
+def test_dataframe_methods_on_accessor(sample_df):
+    xfail_dask_and_koalas(sample_df)
+
+    schema_df = sample_df.copy()
+    schema_df.ww.init(name='test_schema')
+
+    copied_df = schema_df.ww.copy()
+
+    assert schema_df is not copied_df
+    assert isinstance(copied_df.ww.schema, Schema)
+    assert copied_df.ww.schema == schema_df.ww.schema
+
+    pd.testing.assert_frame_equal(to_pandas(schema_df), to_pandas(copied_df))
+
+    error = 'Woodwork typing information is not valid for this DataFrame.'
+    with pytest.raises(ValueError, match=error):
+        schema_df.ww.astype({'id': 'string'})
+
+
+def test_dataframe_methods_on_accessor_inplace(sample_df):
+    xfail_dask_and_koalas(sample_df)
+
+    schema_df = sample_df.copy()
+    schema_df.ww.init(name='test_schema')
+
+    df_pre_sort = schema_df.copy()
+
+    schema_df.ww.sort_values(['full_name'], inplace=True)
+    assert schema_df.ww.name == 'test_schema'
+
+    pd.testing.assert_frame_equal(to_pandas(schema_df), to_pandas(df_pre_sort.sort_values(['full_name'])))
+
+    error = 'Woodwork typing information is not valid for this DataFrame.'
+    with pytest.raises(ValueError, match=error):
+        schema_df.ww.rename({'id': 'new_name'}, inplace=True, axis=1)
+    assert 'new_name' in schema_df.columns
+
+
+def test_dataframe_methods_on_accessor_returning_indexer(sample_df):
+    xfail_dask_and_koalas(sample_df)
+
+    schema_df = sample_df.copy()
+    schema_df.ww.init(name='test_schema')
+
+    iloc_df = schema_df.ww.iloc()[2:, :]
```

Comment:
Want to highlight the behavior of indexers like `iloc`. For some reason, the behavior of the Accessor doesn't allow for `df.ww.iloc[2:, :]`. 

It throws `TypeError: 'function' object is not subscriptable`, probably because of some type transformation that happens when using a getattr like this:
![image](https://user-images.githubusercontent.com/64278226/106632562-8267ff80-6543-11eb-8203-447cc51fd112.png)

My guess is that we might eventually want to add our own `iloc` if this behavior isn't great in the long run. Plus it'd be great to get `df.ww.iloc[:, 2]` to return a series with an initialized Accessor, which we'd definitely need to implement ourselves.

Reply:
Adde #539